cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cluster/ClusteredSortAgentScheduler.java

```diff
@@ -154,6 +154,28 @@ public void schedule(Agent agent, AgentExecution agentExecution, ExecutionInstru
   }
 
   @Override
+  public ClusteredSortAgentLock tryLock(Agent agent) {
+    // This score determines how much time the agent had left in the set it's currently in.
+    String releaseScore = agentScore(agent);
+    String acquireScore = acquireAgent(agent);
```

Comment:
it would be worth making the lua script for acquireAgent return both releaseScore and acquireScore which would make determining these atomic. I think all that would have to change in the lua script is storing the value of the original zrank call and returning it at the end